cmds/core/truncate/truncate.go

```diff
@@ -34,50 +36,38 @@ var (
 
 func init() {
 	flag.Var(size, "s", "Size in bytes, prefixes +/- are allowed")
-
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
-	}
-}
-
-func usageAndExit() {
-	flag.Usage()
-	os.Exit(1)
+	util.Usage(usage)
 }
 
-func main() {
-	flag.Parse()
-
+func truncate(args ...string) error {
 	if !size.IsSet && *rfile == "" {
-		log.Println("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>.")
-		usageAndExit()
+		flag.Usage()
+		return fmt.Errorf("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>")
 	}
 	if size.IsSet && *rfile != "" {
-		log.Println("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>.")
-		usageAndExit()
+		flag.Usage()
+		return fmt.Errorf("truncate: ERROR: You need to specify size via -s <number> or -r <rfile>")
 	}
-	if flag.NArg() == 0 {
-		log.Println("truncate: ERROR: You need to specify one or more files as argument.")
-		usageAndExit()
+	if len(args) == 0 {
+		flag.Usage()
+		return fmt.Errorf("truncate: ERROR: You need to specify one or more files as argument")
 	}
 
-	for _, fname := range flag.Args() {
+	for _, fname := range args {
 
 		var final int64
 		st, err := os.Stat(fname)
 		if os.IsNotExist(err) && !*create {
 			if err = os.WriteFile(fname, []byte{}, 0o644); err != nil {
-				log.Fatalf("truncate: ERROR: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: %v", err)
 			}
 			if st, err = os.Stat(fname); err != nil {
-				log.Fatalf("truncate: ERROR: could not stat newly created file: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: could not stat newly created file: %v", err)
 			}
 		}
 		if *rfile != "" {
 			if st, err = os.Stat(*rfile); err != nil {
-				log.Fatalf("truncate: ERROR: could not stat reference file: %v\n", err)
+				return fmt.Errorf("truncate: ERROR: could not stat reference file: %v", err)
```

Comment:
```
rminnich@t510:~$ truncate -r  /tmp/asdfadfadsffd /tmppppp
truncate: cannot stat '/tmp/asdfadfadsffd': No such file or directory
rminnich@t510:~$
```

It is probably time to start wrapping our errors anyway: 
fmt.Errorf("truncate: %w", err)